### PR TITLE
Redirect to Twitter when it's not Discord calling

### PR DESCRIPTION
**Pull request by adryd325**
Redirect to Twitter without doing any checks when a browser is making a request by checking against known Discord user-agents.

### CODE REVIEW

**Comment by robinuniverse** [comment]
Interesting fix, I'll pull this over to my local test server to give this a go, I don't see a linux useragent so it'll be a good way to see if it works anyways

(Thanks!!)

File: twitfix.py [hunk 1]
```diff
@@ -52,18 +53,22 @@ def oembedend():
...
 
-        res = embedVideo(twitter_url)
-        return res
+            res = embedVideo(twitter_url)
+            return res
+        else:
+            return render_template('default.html', message="This doesn't seem to be a twitter link, try /other/ to see if other kinds of video link works? (experimental)")
     else:
-        return render_template('default.html', message="This doesn't seem to be a twitter link, try /other/ to see if other kinds of video link works? (experimental)")
+        return redirect("https://twitter.com/" + subpath, 301)
```
Comment (robinuniverse) [inline twitfix.py:71]: Subpath is not garunteed to be only the last half of a url, as you can use [host url]/[full twitter url], using it this way causes the redirect to point to a misformatted url (
image
)
Change this line to ``return redirect( twitter_url, 301 )` instead to get the preformatted url
Other than that, I tried it out and it works a lot better!

**Review by robinuniverse — CHANGES_REQUESTED** [review]
Looks good! just needs one small change and I can merge it in :)

**Comment by adryd325** [comment]
> I don't see a linux useragent so it'll be a good way to see if it works anyways
The macOS user agent in the list is an actual user agent that's used when gathering info for embeds, not sure why, but it's been that way for years

File: twitfix.py (hunk 1 above)
Comment (adryd325) [reply twitfix.py:71]: I'll make that fix in the morning, ty!

**Comment by adryd325** [comment]
is this sorta what you want?
```python
@app.route('/<path:subpath>')
def twitfix(subpath):
    user_agent = request.headers.get('user-agent')
    match = pathregex.search(subpath)
    if match is not None:
        twitter_url = subpath

        if match.start() == 0:
            twitter_url = "https://twitter.com/" + subpath

        if user_agent in discord_user_agents:
            res = embedVideo(twitter_url)
            return res
        else:
            return redirect(twitter_url, 301)
    else:
        return redirect("https://twitter.com/" + subpath, 301)
```

**Comment by robinuniverse** [comment]
I can't quite wrap my head around if that bottom redirect will break full urls or not but everything else looks good so I think I'll go ahead and merge it in

**Pull request merged**